sdk/core/core-client-rest/src/helpers/getBinaryBody.ts

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/**
+ * Converts a string representing binary content into a Uint8Array
+ */
+export function stringToBinaryArray(content: string): Uint8Array {
+  const arr = new Uint8Array(content.length);
+  for (let i = 0; i < content.length; i++) {
+    arr[i] = content.charCodeAt(i);
+  }
+
+  return arr;
+}
+
+/**
+ * Converts binary contenty to its string representation
+ */
+export function binaryArrayToString(content: unknown): string {
+  if (typeof content === "string") {
+    return content;
+  }
```

Comment:
as far as I can tell, this if should never get executed, since we always test `ArrayBuffer.isView` or `instanceof Uint8Array`, so we could probably narrow the `content` to be `Uint8Array | ArrayBufferView`, right?